addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoDeviceHandler.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.handler;
+
+import static org.openhab.binding.netatmo.NetatmoBindingConstants.*;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.netatmo.config.NetatmoDeviceConfiguration;
+
+import io.swagger.client.model.NADevice;
+import io.swagger.client.model.NAPlace;
+
+/**
+ * {@link NetatmoDeviceHandler} is the handler for a given
+ * device accessed through the Netatmo Bridge
+ *
+ * @author Gaël L'hopital - Initial contribution OH2 version
+ *
+ */
+public abstract class NetatmoDeviceHandler extends AbstractNetatmoThingHandler {
+    protected NADevice device;
+    private NetatmoDeviceConfiguration configuration;
+
+    public NetatmoDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void bridgeHandlerInitialized(ThingHandler thingHandler, Bridge bridge) {
+        super.bridgeHandlerInitialized(thingHandler, bridge);
+        this.configuration = this.getConfigAs(NetatmoDeviceConfiguration.class);
+
+        scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                updateChannels();
+            }
+        }, 1, configuration.refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    protected void updateChannels() {
+        dashboard = device.getDashboardData();
+
+        super.updateChannels();
+
+        updateConnectedModules();
+    }
+
+    @Override
+    protected State getNAThingProperty(String chanelId) {
+        switch (chanelId) {
+            case CHANNEL_LAST_STATUS_STORE:
+                return new DateTimeType(timestampToCalendar(device.getLastStatusStore()));
+            case CHANNEL_LOCATION:
+                NAPlace place = device.getPlace();
+                return new PointType(new DecimalType(place.getLocation().get(1)),
```

Comment:
@robnielsen @watou besides the argument of losing precision which i am not going to value it simply is unexpected to (potentially) change the value being provided by the Netatmo API. I wouldn't see a reason to change it, too (also my comment on https://github.com/openhab/openhab/pull/3735#discussion_r49254382.